test/integration/targets/blocks/main.yml

```diff
@@ -46,6 +47,18 @@
     #      always_run_after_failure: true
     - meta: clear_host_errors
 
+  # https://github.com/ansible/ansible/issues/35148
+  - block:
+    - block:
+      - name: test triggering always by failing in nested block with run_once set
+        fail:
+      run_once: true
+    always:
+      - name: set block fail always run flag
+        set_fact:
+          nested_block_fail_always: true
+      - meta: clear_host_errors
+
```

Comment:
Would it be useful to add tests here that also sets any_errors_fatal? 

Issues like https://github.com/ansible/ansible/issues/18602 and https://github.com/ansible/ansible/issues/22924 make me think it might be worthwhile. 

(https://github.com/ansible/ansible/pull/25813 adds some vaguely similar test cases focused on any_errors_fatal albeit not directly related to this issue/pr)

Reply:
:+1: